support-frontend/assets/helpers/subscriptionsForms/submit.js

```diff
@@ -80,12 +83,37 @@ function getAddresses(state: AnyCheckoutState) {
 const getOptions = (
   fulfilmentOptions: FulfilmentOptions,
   productOptions: ProductOptions,
-) =>
+): {fulfilmentOptions: FulfilmentOptions, productOptions: ProductOptions} =>
   ({
     ...(fulfilmentOptions !== NoFulfilmentOptions ? { fulfilmentOptions } : {}),
     ...(productOptions !== NoProductOptions ? { productOptions } : {}),
   });
 
+const getProduct =
+  (state: AnyCheckoutState, currencyId?: Option<IsoCurrency>): DigitalSubscription | PaperSubscription => {
+    const {
+      billingPeriod,
+      fulfilmentOption,
+      productOption,
+      orderIsAGift,
+      product,
+    } = state.page.checkout;
+
+    const readerType = orderIsAGift ? Gift : Direct;
+    if (product === DigitalPack) {
+      return {
+        currency: currencyId || state.common.internationalisation.currencyId,
+        billingPeriod,
+        readerType,
```

Comment:
I think we also need the `giftMessage` sent to the backend somewhere around here

Reply:
Yes we do! Thanks John. 🦖

Reply:
thanks for adding - actually I was deliberately vague as I think we should slightly change how the giftee fields arrive at the backend - see my proposal around here: https://github.com/guardian/support-frontend/pull/2730/files#r496931698

Reply:
I am looking at `giftDeliveryDate` now as well, I think that should go into the new DS gift object as well along side the gift message.  This means it won't be optional any more, as if the gift delivery date is required if there is a DS gift.

Reply:
Thanks John, let's make sure it all works correctly after your PR is merged in 👍